test-support/lit-exec-qemu.py

```diff
@@ -26,7 +26,11 @@ def run(args):
     image_args = command[1:]
     qemu_params = args.qemu_params.split(':')
     qemu_cmd = ([args.qemu_command] + qemu_params +
-        ['-semihosting', '-nographic', '-kernel', image])
+        ['-semihosting', '-nographic'])
+    if qemu_params.count("virt"):
```

Comment:
Rather than trying to parse the qemu parameters, pull the machine type out into a separate argument that is parsed with argparse.